pkg/api/types.go

```diff
@@ -173,11 +173,121 @@ type VolumeSource struct {
 	HostPath *HostPath `json:"hostPath"`
 	// EmptyDir represents a temporary directory that shares a pod's lifetime.
 	EmptyDir *EmptyDir `json:"emptyDir"`
+	// GitRepo represents a git repository at a particular revision.
+	GitRepo *GitRepo `json:"gitRepo"`
+
 	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
 	GCEPersistentDisk *GCEPersistentDisk `json:"persistentDisk"`
-	// GitRepo represents a git repository at a particular revision.
-	GitRepo *GitRepo `json:"gitRepo"`
+
+	AWSElasticBlockStore *AWSElasticBlockStore `json:"elasticBlockStore"`
```

Comment:
I think this moves in the wrong direction.  VolumeSource has a meaning and that meaning is "which of the kubelet volume plugins do you want to use".  We want GCE/AWS/... to be less visible in the API.

Instead of adding these here and then re-purposing VolumeSource, make a new struct which represents just the available PersistentVolumeSources.